client/src/main/java/io/pravega/client/netty/impl/ClientConnectionInboundHandler.java

```diff
@@ -128,7 +128,8 @@ public void sendAsync(List<Append> appends, CompletedCallback callback) {
         promise.addListener(new GenericFutureListener<Future<? super Void>>() {
             @Override
             public void operationComplete(Future<? super Void> future) throws Exception {
-                callback.complete(new ConnectionFailedException(future.cause()));
+                Throwable cause = future.cause();
+                callback.complete(cause == null ? null : new ConnectionFailedException(cause));
```

Comment:
It was. This is why this CL was urgent. That was original bug that was reported as #1280, #1307 and #1308